fiasco/gaunt.py

```diff
@@ -0,0 +1,231 @@
+"""
+Gaunt factor object. 
+"""
+import astropy.constants as const
+import astropy.units as u
+import numpy as np
+
+from scipy.ndimage import map_coordinates
+from scipy.special import polygamma
+
+import fiasco
+
+from fiasco.io.datalayer import DataIndexer
+from fiasco.util import check_database, needs_dataset
+
+__all__ = ['GauntFactor']
+
+class GauntFactor():
+    """
+    Class for calculating the Gaunt factor for various continuum processes.
+    """
+    def __init__(self, ion, wavelength=None, freefree=False, freebound=False, 
+                    wavelength_integrated=False,
+                    hdf5_dbase_root=None, *args, **kwargs):
+        if hdf5_dbase_root is None:
+            self.hdf5_dbase_root = fiasco.defaults['hdf5_dbase_root']
+        else:
+            self.hdf5_dbase_root = hdf5_dbase_root
+        check_database(self.hdf5_dbase_root, **kwargs)
+        
+        self.freefree = freefree
+        self.freebound = freebound
+        self.wavelength_integrated = wavelength_integrated
+        
+        self.gf = None
+        if freefree and wavelength is not None and not wavelength_integrated:
+            self.gf = self.free_free(ion, wavelength)
+        elif freefree and wavelength_integrated:
+            self.gf = self.free_free_total(ion)
+        elif freebound and wavelength_integrated:
+            self.gf = self.free_bound_total(ion, **kwargs)
+        
+    def __str__(self):
+        return f"{self.gf}"
+        
+    def __repr__(self):
+        if self.freefree:
+            gaunt_type = "Free-free"
+        elif self.freebound:
+            gaunt_type = "Free-bound"
+        
+        if self.wavelength_integrated:
+            wavelength_type = ", wavelength-integrated"
+        else:
+            wavelength_type = " as a function of wavelength"
+            
+        return f"{gaunt_type} Gaunt Factor{wavelength_type}"
+            
+    @property
+    def _gffgu(self):
+        data_path = '/'.join(['continuum', 'gffgu'])
+        return DataIndexer.create_indexer(self.hdf5_dbase_root, data_path)
+
+    @property
+    def _gffint(self):
+        data_path = '/'.join(['continuum', 'gffint'])
+        return DataIndexer.create_indexer(self.hdf5_dbase_root, data_path)
+    
+    @property
+    def _itohintrel(self):
+        data_path = '/'.join(['continuum', 'itohintrel'])
+        return DataIndexer.create_indexer(self.hdf5_dbase_root, data_path)
+
+    @property
+    def _itohintnonrel(self):
+        data_path = '/'.join(['continuum', 'itohintnonrel'])
+        return DataIndexer.create_indexer(self.hdf5_dbase_root, data_path)
+        
```

Comment:
A bit lost on how to do this.  It looks like I can't just move this over to the `GauntFactor` object and rebuild the database.  (`KeyError: 'continuum/itoh not found in /Users/reep/.fiasco/chianti_dbase.h5'`).  Is there something in the building of the HDF5 file that ties it to the ions?

Reply:
It's because the `_itoh` dataset lives under '{atomic_symbol}/continuum/itoh'. There are two ways we could go about this:

1. (Low effort) Require Z (or some element designation) when instantiating the `GauntFactor` constructor, store it internally, and just copy the current `_itoh` property from `Ion`.
2. (Medium effort) Implement a method on `GauntFactor` that takes in an elemental symbol and returns the appropriate part of that dataset. This requires no changes to the database or parsing code, but does make this data access pattern inconsistent with the rest of the package.
3. (Most effort) Move the Itoh dataset underneath '/continuum/itoh', add the property onto `GauntFactor` in the same manner as the others. Anytime this data is needed, pull out the correct row from that table for the appropriate Z value. This will require passing in Z wherever this is needed. This will also require rebuilding the database after changing the schema.

If we chose option 1, we could also just require passing in an `Ion` object when instantiating `GauntFactor` (in contrast to what I said previously). Maybe that does actually make the most sense?

Reply:
I just implemented Option 3.  Took me a bit to figure out the parser for it, but it's pulling the correct values and is now stored under `GauntFactor`.  

It makes the most sense to me, and a few of the methods need the atomic number anyway.